docs/shared_library/README.md

```diff
@@ -0,0 +1,247 @@
+### Using Scryer Prolog as a Shared Library
+
+Think of a shared library like a toolbox filled with pre-built tools (functions) that other programs can use without needing to rebuild everything from scratch. In our case, `libscryer_prolog.XXX` is the toolbox containing Scryer Prolog's powerful logic engine.
+
+**Note:** Many of the code examples here will be in Python. This is not an endorsement of a particular language, but used for clarity and accessibility, and due to the shared library author's familiarity. 
+
+**Steps to Use the Shared Library:**
+
+1. **Locate the Library:** After building Scryer Prolog, you'll find the shared library in:
+
+   `<PATH-TO>/scryer-prolog/target/release`. 
+   
+   Replace `<PATH-TO>` with the actual path where you installed Scryer Prolog on your computer. The library will have a name like `libscryer_prolog.so` (Linux), `libscryer_prolog.dylib` (macOS), or `scryer_prolog.dll` (Windows).
+
+2. **Load the Library:**  Here are basic examples for common languages:
+
+   * **C/C++ (Linux/macOS):**
+     ```c++
+     #include <dlfcn.h>
+
+     void* handle = dlopen("libscryer_prolog.so", RTLD_LAZY); // Load the library
+     if (!handle) {
+         // Handle error: library not found
+     }
+
+     // ... Get function pointers from the loaded library ...
+     ```
+
+   * **Python:** You'll typically use libraries like `ctypes`:
+      ```python
+      import ctypes
+
+      lib = ctypes.cdll("./libscryer_prolog.so") # Replace with correct path
+
+      # ... Access functions in lib ...
+      ```
+3. **Access Functions:** Once the library is loaded, you can call its functions (like `scryer_consult_module_string`, `scryer_run_query`) just like you would any other function in your code. See the Python reference implementation below.
+
+
+ **Important Notes:**
+
+* **Memory Management:** Be extra careful about memory management when interacting with C libraries from other languages (especially strings). See below.
+
+
+The following functions are exposed:
+
+#### Shared Library Functions Summary
```

Comment:
I think it would make sense to automatically generate such a `scryer.h` file using e.g. [`cbindgen`](https://github.com/mozilla/cbindgen) to ensure the header stays in sync with the rust code.

Reply:
Yes! I thought we would get there eventually, but maybe it's better to start looking into that earlier.